src/UI/WB.UI.Headquarters.Core/Controllers/Api/PublicApi/Graphql/Maps/MapsResolver.cs

```diff
@@ -17,20 +19,82 @@ namespace WB.UI.Headquarters.Controllers.Api.PublicApi.Graphql.Maps
 {
     public class MapsResolver
     {
-        public IQueryable<MapBrowseItem> GetMaps([Service] IUnitOfWork unitOfWork) => 
-            unitOfWork.Session.Query<MapBrowseItem>();
-        
+        public IQueryable<MapBrowseItem> GetMaps([Service] IUnitOfWork unitOfWork, [Service]IAuthorizedUser user)
+        {
+            unitOfWork.DiscardChanges();
+            IQueryable<MapBrowseItem> maps = unitOfWork.Session.Query<MapBrowseItem>();
+            
+            if (user.IsSupervisor)
+            {
+                //add all team maps
+                var team = unitOfWork.Session.Query<HqUser>()   
```

Comment:
remove ToList, add supervisor name in query '''Where'''
In this case all query will be executed on sql side